runtime/src/accounts_index.rs

```diff
@@ -159,15 +159,31 @@ impl<'a, T: 'a + Clone> AccountsIndex<T> {
         reclaims: &mut SlotList<T>,
     ) -> Option<T> {
         if let Some(lock) = self.account_maps.get(pubkey) {
-            let mut list = &mut lock.1.write().unwrap();
-            // filter out other dirty entries
-            reclaims.extend(list.iter().filter(|(f, _)| *f == slot).cloned());
-            list.retain(|(f, _)| *f != slot);
-
-            lock.0.fetch_add(1, Ordering::Relaxed);
+            let list = &mut lock.1.write().unwrap();
+            // filter out other dirty entries from the same slot
+            let mut same_slot_previous_updates: Vec<(usize, (Slot, T))> = list
+                .iter()
+                .enumerate()
+                .filter_map(|(i, (f, value))| {
+                    if *f == slot {
+                        Some((i, (*f, value.clone())))
```

Comment:
nits: maybe this `.clone()` can be moved inside this `if` clause?: https://github.com/solana-labs/solana/pull/12462/files#diff-ef68c28d9d63e66355c44904d9877825R178

i.e. we can `.clone()` only if there is non-empty `same_slot_previous_updates` as a micro-optimization?

Reply:
agreed, updated!